BlazarData/src/main/java/com/hubspot/blazar/data/service/ModuleBuildService.java

```diff
@@ -47,8 +51,32 @@ public ModuleBuildService(ModuleBuildDao moduleBuildDao, ModuleDao moduleDao, Ev
     return moduleBuildDao.getByState(state);
   }
 
-  public List<ModuleBuild> getByModule(int moduleId) {
-    return moduleBuildDao.getByModule(moduleId);
+  public ModuleActivityPage getModuleBuildHistoryPage(int moduleId, Optional<Integer> maybeFromBuildNumber, Optional<Integer> maybePageSize) {
+    int pageSize = MAX_MODULE_ACTIVITY_PAGE_SIZE;
+    if (maybePageSize.isPresent() && 0 < maybePageSize.get() && maybePageSize.get() < MAX_MODULE_ACTIVITY_PAGE_SIZE) {
+      pageSize = maybePageSize.get();
+    }
+
+    int fromBuildNumber;
+    if (maybeFromBuildNumber.isPresent() && 0 < maybeFromBuildNumber.get()) {
+      fromBuildNumber = maybeFromBuildNumber.get();
+    } else {
+      long start = System.currentTimeMillis();
+      Optional<Integer> maxBuildNumber = moduleBuildDao.getMaxBuildNumber(moduleId);
+      LOG.debug("Got max build number for module {} in {}", moduleId, System.currentTimeMillis() - start);
+      if (maxBuildNumber.isPresent()) {
+        fromBuildNumber = maxBuildNumber.get();
+      } else {
+        return new ModuleActivityPage(ImmutableList.of(), 0);
+      }
+    }
+
+    // get a specific page
+    long start = System.currentTimeMillis();
+    List<ModuleBuildInfo> builds = moduleBuildDao.getLimitedModuleBuildHistory(moduleId, fromBuildNumber, pageSize);
```

Comment:
This means we have 3 separate db requests if `fromBuildNumber` is not specified?